src/log.c

```diff
@@ -425,7 +434,6 @@ void qd_vlog_impl(qd_log_source_t *source, qd_log_level_t level, bool check_leve
     }
 
     // Bounded buffer of log entries, keep most recent.
-    sys_mutex_lock(log_lock);
     qd_log_entry_t *entry = new_qd_log_entry_t();
     DEQ_ITEM_INIT(entry);
     entry->module = source->module;
```

Comment:
source->module is a char*. Copying the pointer to entry leaves it vulnerable to deletion as described in the comment at line 446. This string should be copied into entry the same way that entry->file is strdup'd on line 441. Then the entry is not depending on source still existing the whole time entry is in the entries list (line 542).

Reply:
I am proposing that we just move the lock back to where it originally was, like just before this line - qd_log_entry_t *entry = new_qd_log_entry_t();